lib/db/schemaupdater.go

```diff
@@ -484,3 +498,66 @@ func (db *schemaUpdater) updateSchemato9(prev int) error {
 
 	return t.Commit()
 }
+
+func (db *schemaUpdater) updateSchemato10(_ int) error {
```

Comment:
I don't fully understand why we need a migration as opposed to just doing a metadata recalc, which we could force if there is no entry for the need bit for example?

Reply:
This PR adds a `Deleted` flag to the global version, for which a db transition is required. However recalculating needs is indeed not necessary anymore now that we recalc on upgrade. I removed that, which means I redid the transition to be more efficient (unmarshal global lists only once). That also triggered a deficiency in my test data: Didn't set the `prevVersion` to 9 and didn't set any `Sequence` on the file infos -> adjusted generation func and regenerated test db.